CMakeLists.txt

```diff
@@ -151,37 +169,52 @@ function(set_finufft_options target)
         endif ()
     endif ()
 
-    # FFTW CMAKE file includes the APIs only as an install target, so we need to manually
-    # include them since we need them for build not for install
-    # trying to include them directly into the fftw and fftwf targets causes issues with
-    # the latest version of cmake, so we do it here instead.
-    if ((NOT FFTW_FOUND) OR (FINUFFT_FFTW_LIBRARIES STREQUAL DOWNLOAD))
-        list(GET FINUFFT_FFTW_LIBRARIES 0 element)
-        get_property(FFTW_SOURCE_DIR TARGET ${element} PROPERTY SOURCE_DIR)
-        set(FFTW_INCLUDE_DIR ${FFTW_SOURCE_DIR}/api)
-        target_include_directories(${target} PUBLIC ${FFTW_INCLUDE_DIR})
+    if (FINUFFT_USE_DUCC0)
+        target_compile_definitions(${target} PRIVATE FINUFFT_USE_DUCC0)
+    else ()
+        # FFTW CMAKE file includes the APIs only as an install target, so we need to manually
+        # include them since we need them for build not for install
+        # trying to include them directly into the fftw and fftwf targets causes issues with
+        # the latest version of cmake, so we do it here instead.
+        if ((NOT FFTW_FOUND) OR (FINUFFT_FFTW_LIBRARIES STREQUAL DOWNLOAD))
+            list(GET FINUFFT_FFTW_LIBRARIES 0 element)
+            get_property(FFTW_SOURCE_DIR TARGET ${element} PROPERTY SOURCE_DIR)
+            set(FFTW_INCLUDE_DIR ${FFTW_SOURCE_DIR}/api)
+            target_include_directories(${target} PUBLIC ${FFTW_INCLUDE_DIR})
+        endif ()
     endif ()
-    target_link_libraries(${target} PRIVATE xsimd)
-    target_link_libraries(${target} PUBLIC ${FINUFFT_FFTW_LIBRARIES})
+#    target_link_libraries(${target} PRIVATE xsimd)
+#    target_link_libraries(${target} PUBLIC ${FINUFFT_FFTW_LIBRARIES})
 endfunction()
 
 if (FINUFFT_USE_CPU)
     # Main finufft libraries
+    if (FINUFFT_USE_DUCC0)
+        # okay for now, to be cleaned up
+        set_finufft_options(ducc0)
+    endif ()
+
     add_library(finufft_f32 OBJECT ${FINUFFT_PRECISION_DEPENDENT_SOURCES})
     target_compile_definitions(finufft_f32 PRIVATE SINGLE)
     set_finufft_options(finufft_f32)
+    target_link_libraries(finufft_f32 PUBLIC ${FINUFFT_FFTLIBS})
+    target_link_libraries(finufft_f32 PRIVATE xsimd)
 
     add_library(finufft_f64 OBJECT ${FINUFFT_PRECISION_DEPENDENT_SOURCES})
     target_compile_definitions(finufft_f64 PRIVATE)
     set_finufft_options(finufft_f64)
+    target_link_libraries(finufft_f64 PUBLIC ${FINUFFT_FFTLIBS})
+    target_link_libraries(finufft_f64 PRIVATE xsimd)
```

Comment:
@DiamonDinoia Marco, the reason windows can not find xsimd seems to be here, this line is moved out of set_finufft_options function. the below if(WIN32) does not add link xsimd.

Reply:
Great thanks for fixing it!